### ISSUE TITLE
Feedback on `Image Inpainting Autoencoder-decoder approach.ipynb`

### ISSUE THREAD

**Issue by sayakpaul** [ayulockin/deepimageinpainting#1]
- Can't we just select GPU as our runtime and run `%tensorflow_version 2.x` on Colab? 

- Don't pass your `wandb` API key like this: `!wandb login 69f60a7711ce6b8bbae91ac6d15e45d6b1f1430e`. Instead run `wandb.login()`. It will prompt you to supply the API key which you can copy from your `wandb` profile page and paste there. 

- Put a reference to this `createAugment`. If I am not wrong you referred this tutorial: https://stanford.edu/~shervine/blog/keras-how-to-generate-data-on-the-fly. 

- The assignments are spurious:

```
x = y = np.random.randint(0, 31, 1)[0]
w = h = np.random.randint(1, 10, 1)[0]
``` 

- The following is the heart of the data generator:
```
for i, idx in enumerate(idxs):
        tmp_image = self.X[idx].copy()

        mask = np.full(tmp_image.shape, 255, np.uint8)
        mask[y:y+h,x:x+w] = 0
        res = cv2.bitwise_and(tmp_image, mask)

        X_batch[i,] = res/255
        y_batch[i] = self.y[idx]/255
```
Please add self-explanatory comments so that the code easier to read. 

- Regarding the masks, _could they be randomly placed_? 

- I think we can add a utility function for creating the conv and upsamling blocks to avoid repetitions. 

- Include a conceptual plot of the model (use `plot_model`). 

- Why `binary_crossentropy`?

```
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=[dice_coef])
```
